### ISSUE TITLE
Cluster

### ISSUE THREAD

**Issue by misiek08** [openresty/lua-resty-redis#43]
Hi! Is it possible to implement cluster functionality to core of this library? I'm not sure if I am able to do it myself, I tried, but code looks bad and it fails sometimes.

**Comment by wbdu** [comment]
come on

**Comment by bjoe2k4** [comment]
Maybe i'm wrong, but doesn't redis have builtin clustering functionality since version 3.0? Can't we rely on that?

**Comment by misiek08** [comment]
@bjoe2k4 yes it have, but client need to implement few responses from Redis to fully support cluster. Redis have built in only semi-routing which tells client where to ask for data if client connected to wrong node at first.

**Comment by agentzh** [comment]
Guys, there are two pull requests implementing cluster-related features on the client side. Have you checked them out? See

https://github.com/openresty/lua-resty-redis/pulls

**Comment by misiek08** [comment]
I see only one is totally cluster-related but I don't understand a part of code there.... Looks like there are 2 hard-coded limits for cluster count (20) and nodes count in cluster (500).
It looks like this pull is using one object for many cluster and I think that one "client" should be one cluster. About node count limit I think it should be done on dynamically sized array if possible.

**Comment by bungle** [comment]
I will add this to here as a reminder that implementing a cluster support is not something that can be done ligthly, many questions needs to be answered, a lot of design decisions need to be made:
https://github.com/xetorthio/jedis/issues/990#issuecomment-103393965

**Comment by alu** [comment]
+1

**Comment by vkozubal** [comment]
+1

**Comment by spacewander** [comment]
Actually there are some unofficial clients offer part of the redis-cluster support. For example, like this one:
https://github.com/steve0511/resty-redis-cluster
